torch/_dynamo/variables/torch.py

```diff
@@ -335,6 +335,28 @@ def call_function(
             return ConstantVariable(
                 torch.backends.cudnn.is_acceptable(tensor_inp), **options
             )
+        elif self.value is torch.Tensor:
+            # torch.Tensor() calls fail under fake mode.
+            # So, we replace torch.Tensor calls with torch.FloatTensor calls,
+            # or whatever the default tensor type is.
+
+            current_tensor_type = None
+            for tensor_type in torch._tensor_classes:
```

Comment:
Can you explain why we can't fix this on the FakeTensor side?  I worry this isn't fixing the root cause but rather working around gaps in FakeTensor support.  We will also duplicate logic in PyTorch which might be annoying to maintain if we don't get it right.

The ideal fix here would be to fix the root cause with fake tensors.

Alternately, rather than duplicating this dispatching logic (which doesn't seem dynamo specific) -- we could add something like torch.ops.aten.new_tensor(), which mimics the behavior of torch.Tensor() -- assuming such an aten function doesn't already exist.  With that we could either:
1) Make `torch.Tensor.__new__` check for subclassing and call that function if none exists.
2) Or, do a simpler replacement on the dynamo side.

Reply:
@jansel thanks for the input - a few questions:

1) by "fix this on the FakeTensor side" my idea is to add special handling in [`THPVariable_pynew`](https://github.com/pytorch/pytorch/blob/387357539f5933d387656c69cb47aaa85e46ed5f/torch/csrc/autograd/python_variable.cpp#L1805-L1809) to skip `THPVariable_NewWithVar` for plain torch.Tensors separately, if we are in a subclass mode. Does that sound reasonable?
2) for the alternate idea of adding torch.ops.aten.new_tensor, what's the benefit of using aten op over the approach in #92133 , which adds this behavior but not as an aten op?

also added more details in the PR description, starting at `The failure occurs in THPVariable_pynew...`

Reply:
Re (1), this is not quite right, but I think it'll be fastest if I open a PR to show conceptually what to do.

Reply:
https://github.com/pytorch/pytorch/pull/92754

The PR needs a test. Feel free to commandeer and land.

Reply:
I think @ezyang's suggestion is the right way to go.